pkg/provider/provider.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/emicklei/go-restful"
+	"github.com/golang/glog"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/metrics/pkg/apis/custom_metrics"
+
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
+)
+
+type E2EProvider struct {
+	client dynamic.ClientPool
+	mapper apimeta.RESTMapper
+
+	values map[provider.CustomMetricInfo]int64
+}
+
+func NewE2EProvider(client dynamic.ClientPool, mapper apimeta.RESTMapper) provider.CustomMetricsProvider {
+	return &E2EProvider{
+		client: client,
+		mapper: mapper,
+		values: make(map[provider.CustomMetricInfo]int64),
+	}
+}
+
+func (p *E2EProvider) WebService() *restful.WebService {
+	ws := new(restful.WebService)
+
+	ws.Path("/metrics")
```

Comment:
nit: make it more obvious that this isn't part of the normal API pattern (e.g. `write-metrics`, or somesuch).